### ISSUE TITLE
Querying with id instead of _id in MongoDB with Panache

### ISSUE THREAD

**Issue by el10686** [quarkusio/quarkus#26148]
### Description

Suppose we have this very simple class as a Mongo entity with a custom `String` as an `id`
 ```
 @MongoEntity
public class Person {

    private String id;
    private String name;

    // constructors & getters & setters 
```

When querying we have to use `_id` as it is stored in Mongo like:
`find("_id", id)` when it should make sense to use `find("id", id)`

**Comment by loicmathieu** [comment]
@JohnneSouza the PR should be even easier ;)
We already have a mechanism to replace field name in case `@BsonProperty` is used, we can add an hardcoded replacement for `id -> _id`. I don't think it's possible/supported by the POJOCodec to have both `id` and `_id` in a collection/POJO.

**Comment by loicmathieu** [comment]
@JohnneSouza do you still plan to provide a PR for this ?

**Comment by JohnneSouza** [comment]
Hey @loicmathieu sorry, I totally forgot about this thread haha! I'll take a look on it today or tomorrow and push this stuff to the repo ASAP! Tks for the reminder :~D

**Comment by JohnneSouza** [comment]
As always, I'm really forgetting things... but I couldn't help to notice that MongoDB by default will create a new document with the "_id" and I'm think if it wouldn't be a little counter intuitive querying with "id" instead of "_id";
Another thing is:
If you guys think that is OK to change from _id to id, 
Not sure that only changing [here](https://github.com/quarkusio/quarkus/blob/main/extensions/panache/mongodb-panache-common/runtime/src/main/java/io/quarkus/mongodb/panache/common/runtime/MongoOperations.java) would be enough.
@loicmathieu @vapostolopoulos

**Comment by loicmathieu** [comment]
@JohnneSouza I was thinking about this (and an other issue with querying with attributes annotated by `@BsonProperty`. MongoDB PojoCodec is only used to serialize/deserialize a Pojo to/from a collection, at query time MongoDB only takes fields names. 

So if `id` is used instead of `_id`, or if `@BsonProperty` is used, even if it means POJO attributes and collection fields will not be the same it has no effect on the query that must use collection field names.

So I'm not sure allowing someone to use `id` instead of `_id` is a good idea, we should mandates the usage of the field (MongoDB with Panache is not an ORM). I need to check what's our current status around this.

**Comment by evanchooly** [comment]
I'm unclear what that last comment is getting at but, IMO, we should require the use of the mapped names in queries.  Replacing java field names with `@BsonProperty` mapped names gets really tricky without the proper modeling and panache is a bit simplistic in that respect.  My vote is to require the mapped names including `_id`.

**Comment by loicmathieu** [comment]
@evanchooly yes, this is more or less what I'm thinking of, what you named _mapped names_ I named them _collection field names_.

I should update the documentation guide and the JavaDoc to clearly states that when querying, the collection field names should always be used even if `@BsonProperty` is used to change the name as no remapping is done, including `_id`. (I'll come with a better sentence explaining that)
